representatives/contrib/francedata/variants.py

```diff
@@ -86,6 +102,31 @@ def _get_sen_comittees(json):
     return comittees
 
 
+def _get_sen_delegations(json):
+    delegations = []
+    start = json['mandat_debut']
+
+    for g in ([i['responsabilite'] for i in json['responsabilites']] +
+            [j['responsabilite'] for j in json['groupes_parlementaires']]):
+        orga = g['organisme']
+        role = g['fonction']
+
+        if orga.lower().startswith('commission'):
+            continue
+
+        if orga in _equiv_Deleg_SEN:
+            orga = _equiv_Deleg_SEN[orga]
+
+        delegations.append({
+            'abbr': _abbr_Deleg_SEN[orga] if orga in _abbr_Deleg_SEN else '',
+            'name': orga,
+            'role': role,
+            'start': start
+        })
+
+    return delegations
```

Comment:
Perhaps implement a class like:

```
class DelegationGetter(object):
    def __init__(self, equiv, abbr):
        self.equiv, self.abbr = self.equiv, self.abbr
    def __call__(self, data):
         """
         Implement _get_*_delegations() here, use self.abbr and self.equiv instead of _abbr_Deleg_{SEN,AN} _equiv_Deleg_{SEN,AN}
         """
```

Do you think that could work ?